File: clients/rust/src/api/idempotency.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct Idempotency<'a> {
    cfg: &'a Configuration,
}

impl<'a> Idempotency<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Abandon an idempotent request (remove lock without saving response)
    pub async fn abort(
        &self,
        idempotency_abort_in: IdempotencyAbortIn,
    ) -> Result<IdempotencyAbortOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/idempotency/abort")
            .with_body_param(idempotency_abort_in)
            .execute(self.cfg)
            .await
    }

    /// Get idempotency group
    pub async fn get_group(
        &self,
        idempotency_get_group_in: IdempotencyGetGroupIn,
    ) -> Result<IdempotencyGetGroupOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/idempotency/get-group")
            .with_body_param(idempotency_get_group_in)
            .execute(self.cfg)
            .await
    }
}

File: clients/python/coyote/apis/idempotency.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    IdempotencyAbortIn,
    IdempotencyAbortOut,
    IdempotencyGetNamespaceIn,
    IdempotencyGetNamespaceOut,
)


class IdempotencyAsync(ApiBase):
    async def abort(
        self,
        idempotency_abort_in: IdempotencyAbortIn,
    ) -> IdempotencyAbortOut:
        """Abandon an idempotent request (remove lock without saving response)"""
        body = idempotency_abort_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/idempotency/abort",
            body=body,
            response_type=IdempotencyAbortOut,
        )

    async def get_namespace(
        self,
        idempotency_get_namespace_in: IdempotencyGetNamespaceIn,
    ) -> IdempotencyGetNamespaceOut:
        """Get idempotency namespace"""
        body = idempotency_get_namespace_in.model_dump(
            exclude_unset=True, by_alias=True
        )

        return await self._request_asyncio(
            method="post",
            path="/api/v1/idempotency/get-namespace",
            body=body,
            response_type=IdempotencyGetNamespaceOut,
        )


class Idempotency(ApiBase):
    def abort(
        self,
        idempotency_abort_in: IdempotencyAbortIn,
    ) -> IdempotencyAbortOut:
        """Abandon an idempotent request (remove lock without saving response)"""
        body = idempotency_abort_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/idempotency/abort",
            body=body,
            response_type=IdempotencyAbortOut,
        )

    def get_namespace(
        self,
        idempotency_get_namespace_in: IdempotencyGetNamespaceIn,
    ) -> IdempotencyGetNamespaceOut:
        """Get idempotency namespace"""
        body = idempotency_get_namespace_in.model_dump(
            exclude_unset=True, by_alias=True
        )

        return self._request_sync(
            method="post",
            path="/api/v1/idempotency/get-namespace",
            body=body,
            response_type=IdempotencyGetNamespaceOut,
        )
